### ISSUE TITLE
Another dataview problem

### ISSUE THREAD

**Issue by maxless** [kriszyp/msgpackr#21]
The previous one I've had was fixed by the 1.2.3 but this one is new. I've checked it on 1.2.5. The code is attached to the issue. I'm using custom extension to serialize/deserialize linked list objects as arrays and it breaks if there's a list in a list producing this message on unpacking:

```
/home/hexen/node_modules/msgpackr/unpack.js:57
                dataView = source.dataView || (source.dataView = new DataView(source.buffer, source.byteOffset, source.byteLength))
                                                                 ^

TypeError: First argument to DataView constructor must be an ArrayBuffer
    at new DataView (<anonymous>)
    at Object.unpack (/home/hexen/node_modules/msgpackr/unpack.js:57:52)
    at /home/hexen/node_modules/msgpackr/unpack.js:44:24
    at saveState (/home/hexen/node_modules/msgpackr/unpack.js:819:14)
    at Object.unpack (/home/hexen/node_modules/msgpackr/unpack.js:42:11)
    at Array.unpack (/home/hexen/test/nodejs17-msgpack-bug2/main.js:33:27)
    at read (/home/hexen/node_modules/msgpackr/unpack.js:284:38)
    at readMap (/home/hexen/node_modules/msgpackr/unpack.js:481:24)
    at read (/home/hexen/node_modules/msgpackr/unpack.js:337:12)
    at readMap (/home/hexen/node_modules/msgpackr/unpack.js:481:24)
```
[main.txt](https://github.com/kriszyp/msgpackr/files/5990196/main.txt)

**Comment by kriszyp** [comment]
Should be fixed in 1.2.6. Thanks for the great test case!

**Comment by maxless** [comment]
Works fine now, thank you, too :)
